code/modules/hydroponics/seed_mobs.dm

```diff
@@ -57,6 +57,9 @@
 		O = pick(currently_querying)
 		while(currently_querying.len && !check_observer(O)) //While we the list has something and
 			currently_querying -= O				//Remove them from the list if they don't get checked properly
+			if(currently_querying.len) // Are we out of people?
```

Comment:
The `break` breaks the while_loop, which checks for a thing in currently_querying and while O isn't an observer
currently_querying is never added to inside the while block
There is no sleep in the while block for it to wait for new input
`O` is only assigned outside the while loop and within it, if  and it's de-referrenced only if O is picked (with this PR)

The way this loop normally functions is to check if anyone in currently_querying isn't an observer, and then remove them. That's it.